File: sgi-webapp/src/app/core/models/csp/seguimiento-justificacion-anualidad.ts
import { DateTime } from 'luxon';
import { IProyectoPeriodoJustificacionSeguimiento } from './proyecto-periodo-justificacion-seguimiento';

export interface ISeguimientoJustificacionAnualidad {
  proyectoPeriodoJustificacionId: number;
  proyectoId: number;
  proyectoPeriodoJustificacionSeguimiento: IProyectoPeriodoJustificacionSeguimiento;
  identificadorJustificacion: string;
  fechaPresentacionJustificacion: DateTime;
  anio: number;
}

File: sgi-csp-service/src/main/java/org/crue/hercules/sgi/csp/dto/SeguimientoJustificacionAnualidad.java
package org.crue.hercules.sgi.csp.dto;

import java.time.Instant;

import org.crue.hercules.sgi.csp.model.ProyectoPeriodoJustificacionSeguimiento;
import org.crue.hercules.sgi.csp.model.ProyectoProyectoSge;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SeguimientoJustificacionAnualidad {
  /** {@link ProyectoPeriodoJustificacion#id} */
  private Long proyectoPeriodoJustificacionId;
  /** {@link ProyectoPeriodoJustificacionSeguimiento#id} */
  private Long proyectoPeriodoJustificacionSeguimientoId;
  /** {@link ProyectoProyectoSge#proyectoId} */
  private Long proyectoId;
  /** {@link ProyectoPeriodoJustificacion#identificadorJustificacion} */
  private String identificadorJustificacion;
  /** {@link ProyectoPeriodoJustificacion#fechaPresentacionJustificacion} */
  private Instant fechaPresentacionJustificacion;
  /** {@link ProyectoAnualidad#anio} */
  private Integer anio;
}
